### DISCUSSION TITLE
Pull request status of status bar height calculation changes

### DISCUSSION THREAD

**Discussion by ramesh0452** [apache/cordova#397] [General]
Hi, Just wanted to known the status of pull request. Will it be merged to master branch or are there any pending things related to this pull request.
https://github.com/apache/cordova-plugin-statusbar/pull/134

**Comment by breautek** [comment]
It's gotten pretty stale, unfortunately... it was never merged cause usually we want at least two PMC members to give a review. And right now it would need to be rebased.

However if you're trying to use it as a workaround to safe area insets, it's not quite sufficient. It will get you the top area inset, but it won't get you left/right/bottom. And it doesn't appear to give you a listener API should these values ends up changing. While I gave it a +1 in the past, it might really be out of scope of Cordova.

My workplace created a plugin for that problem called [@totalpave/cordova-plugin-insets](https://github.com/totalpaveinc/cordova-plugin-insets)

It's implemented for android-only because iOS has native support for safe area insets. The android plugin offers an API to either get the last known inset information or to listen to inset changes (which will happen on orientation change for example).
It doesn't do anything more then that, so if you want to use the values from CSS like the safe-area-* css values, you'll need to bind them yourself. We didn't do that ourselves because binding CSS variables is incredibly slow, so instead we just adjusted our root app container accordingly. The API is open ended so that you can glue it to fit your needs.
